src/app/navigation/utils.ts

```diff
@@ -109,3 +112,40 @@ export const findPathForParams = (params: object, paths: string[]) => {
       paramsInPath.every(({name}) => paramKeys.includes(name.toString()));
   });
 };
+
+const isScrollTarget = (
+  object: { [key: string]: any }
+): object is HighlightScrollTarget | SearchScrollTarget => {
```

Comment:
lets create a generic ScrollTarget type which has no subtype knowledge, something like
```
interface {
  elementId?: string;
  type: string;
}
```
and at this point only assert that it matches this type. in other places other logic can have other guards for seeing if it matches their expected format